homeassistant/components/rainforest_eagle/sensor.py

```diff
@@ -42,18 +42,27 @@
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
-        vol.Required(CONF_IP_ADDRESS): cv.string,
         vol.Required(CONF_CLOUD_ID): cv.string,
         vol.Required(CONF_INSTALL_CODE): cv.string,
+        vol.Optional(CONF_IP_ADDRESS, default=""): cv.string,
     }
 )
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Create the Eagle-200 sensor."""
-    ip_address = config[CONF_IP_ADDRESS]
     cloud_id = config[CONF_CLOUD_ID]
     install_code = config[CONF_INSTALL_CODE]
+    if config[CONF_IP_ADDRESS] == "":
+        from zeroconf import Zeroconf
```

Comment:
you're not allowed to rely on other integrations requirements. If this is something that we want to add, we should make it a function in the zeroconf integration.

Reply:
got it.  I'll put it in a separate branch which will try to flesh out the config flow and discovery.